lms/djangoapps/teams/views.py

```diff
@@ -877,6 +877,14 @@ def post(self, request):
         except User.DoesNotExist:
             return Response(status=status.HTTP_404_NOT_FOUND)
 
+        course_module = modulestore().get_course(team.course_id)
+        if ('max_team_size' in course_module.teams_configuration and
+            team.users.count() >= course_module.teams_configuration['max_team_size']):
+            return Response(
+                build_api_error(ugettext_noop("The team is already full.")),
+                status=status.HTTP_400_BAD_REQUEST
```

Comment:
Still going back and forth on the error code, since this really isn't a "bad request" in the case of the original bug.